src/gui/editors/SongEditor.cpp

```diff
@@ -307,42 +297,40 @@ void SongEditor::loadSettings( const QDomElement& element )
 
 
 
+/*! \brief Return grid size as number of bars */
 float SongEditor::getSnapSize() const
 {
-	// 1 Bar is the third value in the snapping dropdown
-	int val = -m_snappingModel->value() + 3;
+	float snapSize = SNAP_SIZES[m_snappingModel->value()];
+
 	// If proportional snap is on, we snap to finer values when zoomed in
 	if (m_proportionalSnap)
 	{
-		val = val - static_cast<int>(round(log2(m_zoomingLogValue / ZOOM_MIN))) + 3;
+		// Finds the closest available snap size
+		const float optimalSize = snapSize * DEFAULT_PIXELS_PER_BAR / pixelsPerBar();
+		return *std::min_element(SNAP_SIZES.begin(), SNAP_SIZES.end(), [&](float a, float b)
```

Comment:
It's not pretty but it would support snap sizes that aren't powers of 2.... (Or if we leave out snap sizes like "2 bars")